### ISSUE TITLE
2 implementation details: self.reg_branches' role in PETRTransformer and seemingly meaningless calculation of query_embeds

### ISSUE THREAD

**Issue by xXuHaiyang** [megvii-research/PETR#108]
Hello! I wanna ask two small implementation details which confuse me. 

1. [forward input of PETRTransformer](https://github.com/megvii-research/PETR/blob/main/projects/mmdet3d_plugin/models/dense_heads/petr_head.py#L419)
    `outs_dec, _ = self.transformer(x, masks, query_embeds, pos_embed, self.reg_branches)`
    I can't figure out what self.reg_branches does in self.transformer, and I tried to delete it, and the experiment output seems to be OK. Could you please give the meaning of adding self.reg_branches to self.transformer? 

2. `reference_points = self.reference_points.weight`
    `query_embeds = self.query_embedding(pos2posemb3d(reference_points))`
    `self.query_embedding = nn.Sequential(
            nn.Linear(self.embed_dims*3//2, self.embed_dims),
            nn.ReLU(),
            nn.Linear(self.embed_dims, self.embed_dims),
        )`
     We get query_embed by above lines, right? But here, [forward procedure of PETRTransformer](https://github.com/megvii-research/PETR/blob/main/projects/mmdet3d_plugin/models/utils/petr_transformer.py#L96), the query_embed is set to zero and feed to the decoder here [query=target](https://github.com/megvii-research/PETR/blob/main/projects/mmdet3d_plugin/models/utils/petr_transformer.py#L100). So why do we need to process the query_embed with a MLP? What's the difference between it and directly create query_embeds with the channel of self.embed_dims and filled by zeros? Could you please give the meaning of passing it through self.query_embedding? 

Thanks!

**Comment by yingfei1016** [comment]
Hello!

1、self.reg_branches
In 2D detection, many networks have this branch to refine bbox layer by layer. But in our practice, It doesn't improve the performance in 3D multi-view detection. So we left this parameter and didn't implement it

2、In DETR, the query consists of **context embedding (query or target)** and **position embedding (query_pos)**.  **context embedding** will added with the **position embedding** to generate query in decoder layer. The context embedding of vanilla DETR is alaways Zero initialization. We follow anchor DETR to generate the position embedding by anchor points (reference_points).

**Comment by xXuHaiyang** [comment]
Thanks a lot! 
I see.
